Sources/Identity/IdentityManager.swift

```diff
@@ -39,27 +41,51 @@ class IdentityManager: CurrentUserProvider {
         backend: Backend,
         customerInfoManager: CustomerInfoManager,
         attributeSyncing: AttributeSyncing,
-        appUserID: String?
+        appUserID: String?,
+        cloudSyncedAnonymousIDMode: Bool = false
     ) {
         self.deviceCache = deviceCache
         self.backend = backend
         self.customerInfoManager = customerInfoManager
         self.attributeSyncing = attributeSyncing
+        self.cloudSyncedAnonymousIDMode = cloudSyncedAnonymousIDMode
 
         if appUserID?.isEmpty == true {
             Logger.warn(Strings.identity.logging_in_with_empty_appuserid)
         }
 
-        let appUserID = appUserID?.notEmptyOrWhitespaces
+        let finalAppUserID: String
+
+        if cloudSyncedAnonymousIDMode {
+            finalAppUserID = appUserID?.notEmptyOrWhitespaces
+              ?? cloudSyncedAnonymousIDProvider.appUserID
+        } else {
+            finalAppUserID = appUserID?.notEmptyOrWhitespaces
             ?? deviceCache.cachedAppUserID
             ?? deviceCache.cachedLegacyAppUserID
             ?? Self.generateRandomID()
+        }
 
         Logger.user(Strings.identity.identifying_app_user_id)
 
-        deviceCache.cache(appUserID: appUserID)
+        deviceCache.cache(appUserID: finalAppUserID)
         deviceCache.cleanupSubscriberAttributes()
-        self.invalidateCachesIfNeeded(appUserID: appUserID)
+        self.invalidateCachesIfNeeded(appUserID: finalAppUserID)
+
+        // todo: inject store and notification center
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(self.ubiquitousKeyValueStoreDidChange),
+            name: NSUbiquitousKeyValueStore.didChangeExternallyNotification,
```

Comment:
Should we only subscribe to this if in `cloudSyncedAnonymousIDMode`? Maybel also the `applicationWillEnterForegroundNotification` subscribe since right now it's only for this use case as well